src/lib/sed.c

```diff
@@ -185,13 +185,11 @@ int  sed_level0_discovery(struct sed_opal_level0_discovery *discv)
 
 void sed_deinit(struct sed_device *dev)
 {
-	if (dev != NULL) {
+	if (dev->fd)
```

Comment:
this function will crash if it will be called in following way:
sed_deint(NULL)